apps/examples/testcase/le_tc/network/tc_net_poll.c

```diff
@@ -0,0 +1,244 @@
+
+/*  @file tc_net_poll.c
+ * @brief Test Case Example for lwip_poll() API
+ */
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <poll.h>
+#include <pthread.h>
+#include <sys/ioctl.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <netutils/netlib.h>
+#include <lwip/sockets.h>
+#include <tinyara/os_api_test_drv.h>
+#include "tc_internal.h"
+
+struct poll_s {
+	int fd;
+	int sd;
+};
+
+/**
+* @testcase             :tc_net_poll_p
+* @brief                :check for poll in multi-thread env
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_multi_thread_poll_p(int ret, int sd)
+{
+	TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+	TC_SUCCESS_RESULT();
+}
+
+void *wait_event(void *arg)
+{
+	struct pollfd fds[1];
+	struct poll_test_args args;
+	struct poll_s *poll = (struct poll_s *)arg;
+	int sd = poll->sd;
+	int fd = poll->fd;
+	int ret = -1;
+
+	fds[0].fd = sd;
+	fds[0].events = POLLIN;
+
+	args.fd = sd;
+	args.fds = fds;
+	args.setup = 1;
+
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	tc_net_multi_thread_poll_p(ret, sd);
+
+	return NULL;
+}
+
+void *multi_thread_poll(void *arg)
+{
+	int sd = -1;
+	int fd = *(int *)arg;
+	pthread_t thread1;
+	struct poll_s args;
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sd < 0) {
+		printf("socket fail %s:%d", __FUNCTION__, __LINE__);
+		return NULL;
+	}
+
+	args.fd = fd;
+	args.sd = sd;
+	pthread_create(&thread1, NULL, wait_event, (void *)&args);
+
+	close(sd);
+
+	return NULL;
+}
+
+/**
+* @testcase             :tc_net_poll_p
+* @brief                :check for poll
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_p(int fd)
+{
+	int ret = -1;
+	int sd = -1;
+	struct pollfd fds[1];
+	struct poll_test_args args;
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+
+	fds[0].fd = sd;
+	fds[0].events = POLLIN;
+
+	args.fd = sd;
+	args.fds = fds;
+	args.setup = 1;
+
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+
+	args.setup = 0;
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd));
+
+	close(sd);
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             :tc_net_poll_fds_n
+* @brief                :test with wrong values of fds.
+* @scenario             :
+* @apicovered           :lwip_poll()
+* @precondition         :
+* @postcondition        :
+*/
+
+static void tc_net_poll_fds_n(int fd)
+{
+	int ret = -1;
+	int sd = -1;
+	struct poll_test_args args;
+
+	sd = socket(AF_INET, SOCK_DGRAM, 0);
+	if (sd < 0) {
+		printf("socket fail %s:%d", __FUNCTION__, __LINE__);
+		return;
+	}
+
+	args.fd = sd;
+	args.fds = NULL;
+	args.setup = 1;
+
+	ret = ioctl(fd, TESTIOC_NET_POLL, &args);
+	TC_ASSERT_LEQ_CLEANUP("lwip_poll", ret, -1, close(sd));
```

Comment:
TC_ASSERT_LEQ_CLEAN_UP is right? I think TC_ASSERT_GEQ_CLEANUP("lwip_poll", ret, 0, close(sd)); is right.

Reply:
Oh, I missed negative test. I think TC_ASSERT_EQ_CLEANUP("lwip_poll", ret, -1, close(sd)); is right.
That's because ioctl should return -1 in this case.